transaction.go

```diff
@@ -290,6 +291,91 @@ func (t *Transaction) SignalUpdatesApplied() error {
 	return nil
 }
 
+// append is a helper function to append updates to a transaction on which
+// SignalSetupComplete hasn't been called yet
+func (t *Transaction) append(updates []Update, done chan error) {
+	defer close(done)
+
+	// If there is nothing to append we are done
+	if len(updates) == 0 {
+		return
+	}
+
+	// Make sure that the initialization finished
+	err := <-t.initComplete
```

Comment:
We now have 2 different places where we wait on this channel. Unfortunately, the error is only sent down the channel once, meaning if one place catches an error, the other place will either deadlock or see `nil`.

So we have to change our `initComplete` model. My feeling is that we should still use a channel, but the channel is just to indicate that the error has been set. After we see the error has been set, we can read it in parallel without risk because there will be no more writing to that field. So we don't need locking there.

Reply:
The way it is implemented at the moment the other place will see `nil`. Since a `transaction` is not used in parallel I thought the caller would not signal a completed setup when `Append` returns an error.

Reply:
I was wondering about that too; I assume that the channel would be closed. I agree that the channel should signal that the error has been set, and afterward callers can read `t.initErr` (or whatever) without risk of racing.

Reply:
ok done